lib/fs.js

```diff
@@ -196,13 +196,76 @@ function Stats(
   this.ino = ino;
   this.size = size;
   this.blocks = blocks;
-  this.atime = new Date(atim_msec);
-  this.mtime = new Date(mtim_msec);
-  this.ctime = new Date(ctim_msec);
-  this.birthtime = new Date(birthtim_msec);
+  this._atim_msec = atim_msec;
+  this._mtim_msec = mtim_msec;
+  this._ctim_msec = ctim_msec;
+  this._birthtim_msec = birthtim_msec;
 }
 fs.Stats = Stats;
 
+Object.defineProperties(Stats.prototype, {
+  atime: {
+    configurable: true,
+    enumerable: true,
+    get() {
+      return this._atime !== undefined ?
+          this._atime :
+          (this._atime = new Date(this._atim_msec + 0.5));
+    },
```

Comment:
Can eliminate some code duplication here by having a factory function for the getters... e.g.

```js
function makeGetter(name, src) {
  return function() {
    return this[name] !== undefined ?
      this[name] : (this[name] = new Date(this[src + 0.5));
  };
}

// ...

Object.defineProperties(Stats.prototype, {
  atime: {
    configurable: true,
    enumerable: true,
    get: makeGetter('_atime', '_atim_msec'),
    // etc
  }
});
```